src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift

```diff
@@ -219,16 +219,29 @@ final class SecureStore: SecureKeyValueStoring, Store, AntigenTestProfileStoring
 
     // MARK: - Protocol AntigenTestProfileStoring
 
-	private(set) lazy var antigenTestProfileSubject = CurrentValueSubject<AntigenTestProfile?, Never>(antigenTestProfile)
-
-	var antigenTestProfile: AntigenTestProfile? {
-		get { kvStore["antigenTestProfile"] as AntigenTestProfile? }
+	private(set) lazy var antigenTestProfilesSubject = CurrentValueSubject<[AntigenTestProfile], Never>(antigenTestProfiles)
+		
+	var antigenTestProfiles: [AntigenTestProfile] {
+		get {
+			var antigenTestProfiles = kvStore["antigenTestProfiles"] as [AntigenTestProfile]? ?? []
+			
+			if let existingProfile = kvStore["antigenTestProfile"] as AntigenTestProfile? {
+				// add existing profile to the list of profiles and update store
+				antigenTestProfiles.append(existingProfile)
+				kvStore["antigenTestProfiles"] = antigenTestProfiles
+				
+				// clear the existing profile from the store
+				kvStore["antigenTestProfile"] = nil
```

Comment:
Not sure if it will work every time.
Szenario: 'antigenTestProfiles' was get, but never set. So only a read operation happened. Then the old profile would get lost?

Reply:
If the antigenTestProfiles was never set then antigenTestProfiles will be empty array and if an existing profile is there, will add that profile into the antigenTestProfiles and then set the store in line 231. So, the old profile is retained.

Reply:
Correct, somehow I missed line 231